src/Process/Runner/Parallel/ParallelProcessRunner.php

```diff
@@ -75,57 +74,55 @@ public function __construct(EventDispatcher $eventDispatcher)
      * @throws RuntimeException
      * @throws LogicException
      */
-    public function run(array $processes, int $threadCount, int $poll = 1000): void
+    public function run(iterable $processes, int $threadCount, int $poll = 1000): void
```

Comment:
any reason why we have `iterable` here? I'm not too fond of it because it's really a wide and tricky type: you can fit arrays, traversables, generators (rewindable or not)

Reply:
Yes, the reason is in the making. It is a great type because it fits arrays, unlike generators, which makes a breeze mocking methods typically returning a generator.

Reply:
But why do we need to expand the type if we can use only arrays which is already easy to mock?